src/Features/LanguageServer/Protocol/Handler/CodeActions/PreviewCodeActionsHandler.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.LanguageServer.CustomProtocol;
+using Microsoft.CodeAnalysis.PooledObjects;
+using LSP = Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler
+{
+    [Shared]
+    [ExportLspMethod(RoslynMethods.CodeActionPreviewName)]
+    internal class PreviewCodeActionsHandler : CodeActionsHandlerBase, IRequestHandler<RunCodeActionParams, LSP.TextEdit[]>
+    {
+        [ImportingConstructor]
+        public PreviewCodeActionsHandler(ICodeFixService codeFixService, ICodeRefactoringService codeRefactoringService)
+            : base(codeFixService, codeRefactoringService)
+        {
+        }
+
+        public async Task<LSP.TextEdit[]> HandleRequestAsync(Solution solution, RunCodeActionParams request,
+            LSP.ClientCapabilities clientCapabilities, CancellationToken cancellationToken)
+        {
+            var edits = ArrayBuilder<LSP.TextEdit>.GetInstance();
+
+            var codeActions = await GetCodeActionsAsync(solution,
+                                                        request.TextDocument.Uri,
+                                                        request.Range,
+                                                        cancellationToken).ConfigureAwait(false);
+
+            var actionToRun = codeActions?.FirstOrDefault(a => a.Title == request.Title);
+
+            if (actionToRun != null)
+            {
+                var operations = await actionToRun.GetOperationsAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
hmm... this should use GetPreviewOperation.... 

probably we want work item to take a look. for now, let's take live share's impementatino as it is.

Reply:
Sure! https://github.com/dotnet/roslyn/projects/45#card-20866385